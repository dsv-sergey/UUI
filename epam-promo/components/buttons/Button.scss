@use  '../../assets/styles/index' as *;

.root {
    @include button-reset();

    &:global(.uui-button-box) {
        border-style: solid;
        justify-content: center;
    }

    :global(.uui-caption) {
        @include overflow-ellipsis();
        min-width: 0;
        text-align: center;
        flex: 1 1 auto;
        font-family: $font-sans-semibold;
        font-weight: normal;
        user-select: none;
    }
}

@mixin button-disables-styles($bg-color, $border-color) {
    background-color: $bg-color;
    color: var(--caption-color-disabled);
    fill: var(--caption-color-disabled);
    border-color: $border-color;
}

.fill-solid {
    background-color: var(--background-color);
    color: var(--background-color-contrast);
    fill: var(--background-color-contrast);
    border-color: var(--border-color);

    &:global(.-clickable) {
        &:hover {
            background-color: var(--background-color-hover);
            border-color: var(--background-color-hover);
        }

        &:active {
            background-color: var(--background-color-active);
            border-color: var(--background-color-active);
        }
    }

    &:global(.uui-disabled) {
        @include button-disables-styles(var(--background-color-disabled), var(--background-color-disabled));
    }

    &:global(.-clickable), &:hover, &:active, &:focus {
        &:global(.uui-disabled) {
            @include button-disables-styles(var(--background-color-disabled), var(--background-color-disabled));
        }
    }
}

.fill-white {
    background-color: white;
    color: var(--caption-color);
    fill: var(--caption-color);
    border-color: var(--border-color);

    &:global(.-clickable) {
        &:hover {
            background-color: var(--background-color-light-hover);
            color: var(--caption-color);
            fill: var(--caption-color);
            border-color: var(--border-color);
        }

        &:active {
            background-color: var(--background-color-light-active);
            color: var(--caption-color);
            fill: var(--caption-color);
            border-color: var(--border-color);
        }
    }

    &:global(.uui-disabled) {
        @include button-disables-styles(white, var(--background-color-disabled));
    }

    &:global(.-clickable), &:hover, &:active, &:focus {
        &:global(.uui-disabled) {
            @include button-disables-styles(white, var(--background-color-disabled));
        }
    }
}

.fill-none {
    background-color: transparent;
    color: var(--caption-color);
    fill: var(--caption-color);
    border-color: var(--border-color);

    &:global(.-clickable) {
        &:hover {
            background-color: var(--background-color-light-hover);
            color: var(--caption-color);
            fill: var(--caption-color);
            border-color: var(--border-color);
        }

        &:active {
            background-color: var(--background-color-light-active);
            color: var(--caption-color);
            fill: var(--caption-color);
            border-color: var(--border-color);
        }
    }

    &:global(.uui-disabled) {
        @include button-disables-styles(transparent, var(--background-color-disabled));
    }

    &:global(.-clickable), &:hover, &:active, &:focus {
        &:global(.uui-disabled) {
            @include button-disables-styles(transparent, var(--background-color-disabled));
        }
    }
}

.fill-light {
    background-color: transparent;
    color: var(--background-color);
    fill: var(--background-color);
    border-color: transparent;

    &:global(.-clickable) {
        &:hover {
            background-color: var(--background-color-light-hover);
            border-color: var(--background-color-light-hover);
        }

        &:active {
            background-color: var(--background-color-light-active);
            border-color: var(--background-color-light-active);
        }
    }


    &:global(.uui-disabled) {
        @include button-disables-styles(transparent, transparent);
    }


    &:global(.-clickable), &:hover, &:active, &:focus {
        &:global(.uui-disabled) {
            @include button-disables-styles(transparent, transparent);
        }
    }
}

//Sizes
.size-24 {
    @include button-layout(24px, 1px);
}

.size-30 {
    @include button-layout(30px, 1px);
}

.size-36 {
    @include button-layout(36px, 1px);
}

.size-42 {
    @include button-layout(42px, 1px);

    :global(.uui-caption) {
        font-family: $font-museo-sans;
        text-transform: uppercase;
    }
}

.size-48 {
    @include button-layout(48px, 1px);

    :global(.uui-caption) {
        font-family: $font-museo-sans;
        text-transform: uppercase;
    }
}
