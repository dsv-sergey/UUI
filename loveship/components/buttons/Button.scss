@use '../../assets/styles/scss/index' as *;

.root {
    @include button-reset();
    font-family: $font-sans-semibold;
    font-weight: normal;

    &:global(.uui-button-box) {
        border-style: solid;
        justify-content: center;
        border-radius: $control-border-radius;
    }

    :global(.uui-caption) {
        @include overflow-ellipsis();
        min-width: 0;
        text-align: center;
        flex: 1 1 auto;
    }

    &:global(.-clickable):focus {
        @include control-focus-effect();
    }
}

// Sizes
.size-18 { @include button-layout(18px, 1px); }
.size-24 { @include button-layout(24px, 1px); }
.size-30 { @include button-layout(30px, 1px); }
.size-36 { @include button-layout(36px, 1px); }
.size-42 { @include button-layout(42px, 1px); }
.size-48 { @include button-layout(48px, 1px); }

// Border style
.style-round {
    &.size-48 { border-radius: 24px; }
    &.size-42 { border-radius: 21px; }
    &.size-36 { border-radius: 18px; }
    &.size-30 { border-radius: 15px; }
    &.size-24 { border-radius: 12px; }
    &.size-18 { border-radius: 10px; }
}

.fill-solid {
    @include button-colors(var(--uui-color-700), var(--uui-contrast-color-700), var(--uui-color-700));

    &:global(.-clickable) {
        &:hover {
            @include button-colors(var(--uui-color-800), var(--uui-contrast-color-700), var(--uui-color-800));
        }

        &:active {
            @include button-colors(var(--uui-color-900), var(--uui-contrast-color-700), var(--uui-color-900));
        }
    }

    &:global(.uui-disabled) {
        @include button-colors($night300, $night500, $night300);
        cursor: default;

        &:hover, &:active, &:focus {
            @include button-colors($night300, $night500, $night300);
            box-shadow: none;
        }

        &::placeholder {
            color: $night300;
        }
    }
}

.fill-white {
    @include button-colors(white, var(--uui-color-700), var(--uui-color-700));

    &:global(.-clickable) {
        &:hover {
            @include button-colors(var(--uui-color-500), var(--uui-color-800), var(--uui-color-800));
        }

        &:active {
            @include button-colors(var(--uui-color-600), var(--uui-color-900), var(--uui-color-900));
        }
    }

    &:global(.uui-disabled) {
        @include button-colors(white, $night500, $night300);
        cursor: default;

        &:hover, &:active, &:focus {
            @include button-colors(white, $night500, $night300);
            box-shadow: none;
        }

        &::placeholder {
            color: $night300;
        }
    }
}

.fill-none {
    @include button-colors(transparent, var(--uui-color-700), var(--uui-color-700));

    &:global(.-clickable) {
        &:hover {
            @include button-colors(var(--uui-color-500), var(--uui-color-800), var(--uui-color-800));
        }

        &:active {
            @include button-colors(var(--uui-color-600), var(--uui-color-900), var(--uui-color-900));
        }
    }

    &:global(.uui-disabled) {
        @include button-colors(transparent, $night500, $night300);
        cursor: default;

        &:hover, &:active, &:focus {
            @include button-colors(transparent, $night500, $night300);
            box-shadow: none;
        }

        &::placeholder {
            color: $night300;
        }
    }
}

.fill-light {
    @include button-colors(transparent, var(--uui-color-700), transparent);

    &:global(.-clickable) {
        &:hover {
            @include button-colors(var(--uui-color-500), var(--uui-color-800), var(--uui-color-500));
        }

        &:active {
            @include button-colors(var(--uui-color-600), var(--uui-color-900), var(--uui-color-600));
        }
    }

    &:global(.uui-disabled) {
        @include button-colors(transparent, $night500, transparent);
        cursor: default;

        &:hover, &:active, &:focus {
            @include button-colors(transparent, $night500, transparent);
            box-shadow: none;
        }

        &::placeholder {
            color: $night300;
        }
    }
}
