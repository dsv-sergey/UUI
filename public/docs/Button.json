{"object":"value","document":{"object":"document","data":{},"nodes":[{"object":"block","type":"uui-richTextEditor-left","data":{},"nodes":[{"object":"block","type":"align-center","data":{},"nodes":[{"object":"text","text":"","marks":[]}]}]},{"object":"block","type":"paragraph","data":{},"nodes":[{"object":"text","text":"","marks":[{"object":"mark","type":"uui-richTextEditor-bold","data":{}},{"object":"mark","type":"uui-richTextEditor-italic","data":{}}]}]}]}}