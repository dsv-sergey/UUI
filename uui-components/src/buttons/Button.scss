.container {
    display: flex;
    position: relative;

    &:global(.-clickable) {
        cursor: pointer;
    }

    &:global(.uui-disabled) {
        cursor: default;
    }
}