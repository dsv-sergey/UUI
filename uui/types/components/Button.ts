import {IBasicPickerToggler, ICanRedirect, IClickable, IDisableable, IDropdownToggler, IHasCaption, IHasCX, IHasIcon, IHasPlaceholder, IAnalyticableClick } from "../props";
import {Icon} from "../objects";

export interface ButtonBaseCoreProps extends IHasCX, IClickable, ICanRedirect, IDisableable, IHasIcon, IAnalyticableClick {
    isLinkActive?: boolean;
    tabIndex?: number;
}

export interface ButtonCoreProps extends ButtonBaseCoreProps, IHasCaption, IBasicPickerToggler, IDropdownToggler, IHasPlaceholder {
    dropdownIcon?: Icon;
    dropdownIconPosition?: 'left' | 'right';
    countPosition?: 'left' | 'right';
    count?: number | null;
}

export interface ButtonSemanticProps {
    type?: 'success' | 'cancel' | 'action';
}