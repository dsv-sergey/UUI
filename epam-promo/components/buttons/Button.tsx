import { Button as uuiButton, ButtonProps } from '@epam/uui-components';
import { withMods } from '@epam/uui';
import { ControlSize, FillStyle } from '../types';
import { systemIcons } from '../../icons/icons';
import * as css from './Button.scss';
import * as styles from '../../assets/styles/colorvars/buttons/button-colorvars.scss';

export type ButtonColor = 'blue' | 'green' | 'red' | 'gray50';
export const allButtonColors: ButtonColor[] = ['blue', 'green', 'red', 'gray50'];

const defaultSize = '36';

export interface ButtonMods {
    size?: ControlSize | '18';
    fill?: FillStyle;
    color?: ButtonColor;
}

export function applyButtonMods(mods: ButtonMods & ButtonProps) {
    return [
        styles['button-color-' + (mods.color || 'blue')],
        css.root,
        css['size-' + (mods.size || defaultSize)],
        css['fill-' + (mods.fill || 'solid')],
    ];
}

export const Button = withMods<ButtonProps, ButtonMods>(
    uuiButton,
    applyButtonMods,
    (props) => ({
        dropdownIcon: systemIcons[props.size || defaultSize].foldingArrow,
        clearIcon: systemIcons[props.size || defaultSize].clear,
    }),
);