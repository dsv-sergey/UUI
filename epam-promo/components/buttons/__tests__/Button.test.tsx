import React from 'react';
import { Button } from '../Button';
import renderer from 'react-test-renderer';
import * as calendarIcon from '../../../icons/calendar-18.svg';
import * as clearIcon from '@epam/assets/icons/common/navigation-close-18.svg';


describe('Button', () => {
    it('should be rendered correctly', () => {

        const tree = renderer
            .create(<Button />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly with props', () => {

        const tree = renderer
            .create(<Button
                color='blue'
                fill='white'
                caption='Click me'
                onClick={ jest.fn }
                icon={ calendarIcon }
                isDisabled={ false }
                isDropdown={ true }
                count={ 10 }
                iconPosition='right'
                onClear={ jest.fn }
                isOpen
                clearIcon={ clearIcon }
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});