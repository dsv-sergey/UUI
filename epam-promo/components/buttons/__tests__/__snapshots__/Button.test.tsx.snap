// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should be rendered correctly 1`] = `
<div
  className="container uui-button-box uui-enabled button-color-blue root size-36 fill-solid"
  href={null}
  onClick={[Function]}
  onKeyDown={[Function]}
  tabIndex={0}
/>
`;

exports[`Button should be rendered correctly with props 1`] = `
<div
  className="container uui-button-box uui-enabled -clickable button-color-blue root size-36 fill-white"
  href={null}
  onClick={[Function]}
  onKeyDown={[Function]}
  tabIndex={0}
>
  <div
    className="uui-count"
  >
    10
  </div>
  <div
    className="uui-caption"
  >
    Click me
  </div>
  <div
    className="container uui-icon uui-enabled"
  >
    <svg
      className=""
      height={24}
      viewBox="0 0 12 24"
      width={12}
    >
      <use
        xlinkHref="#calendar-18.svg"
      />
    </svg>
  </div>
  <div
    className="container uui-icon uui-enabled"
  >
    <svg
      className="flipY"
      height={24}
      viewBox="0 0 12 24"
      width={12}
    >
      <use
        xlinkHref="#folding-arrow-18.svg"
      />
    </svg>
  </div>
  <div
    className="container uui-icon uui-enabled -clickable -clickable"
    onClick={[Function]}
  >
    <svg
      className=""
      height={24}
      viewBox="0 0 12 24"
      width={12}
    >
      <use
        xlinkHref="#btn-cross-18.svg"
      />
    </svg>
  </div>
</div>
`;
